catalog/rocks/entry_catalog.go

```diff
@@ -140,52 +200,138 @@ func (e *EntryCatalog) ListTags(ctx context.Context, repositoryID graveler.Repos
+		return nil, err
+	}
 	return e.store.Log(ctx, repositoryID, commitID)
 }
 
 func (e *EntryCatalog) ListBranches(ctx context.Context, repositoryID graveler.RepositoryID) (graveler.BranchIterator, error) {
+	if err := Validate("repositoryID", repositoryID, ValidateRepositoryID); err != nil {
+		return nil, err
+	}
 	return e.store.ListBranches(ctx, repositoryID)
 }
 
 func (e *EntryCatalog) DeleteBranch(ctx context.Context, repositoryID graveler.RepositoryID, branchID graveler.BranchID) error {
+	if err := Validate("repositoryID", repositoryID, ValidateRepositoryID); err != nil {
+		return err
+	}
+	if err := Validate("branchID", branchID, ValidateBranchID); err != nil {
+		return err
+	}
 	return e.store.DeleteBranch(ctx, repositoryID, branchID)
 }
 
 func (e *EntryCatalog) Commit(ctx context.Context, repositoryID graveler.RepositoryID, branchID graveler.BranchID, committer string, message string, metadata graveler.Metadata) (graveler.CommitID, error) {
+	if err := Validate("repositoryID", repositoryID, ValidateRepositoryID); err != nil {
+		return "", err
+	}
+	if err := Validate("branchID", branchID, ValidateBranchID); err != nil {
+		return "", err
+	}
 	return e.store.Commit(ctx, repositoryID, branchID, committer, message, metadata)
 }
 
 func (e *EntryCatalog) GetCommit(ctx context.Context, repositoryID graveler.RepositoryID, commitID graveler.CommitID) (*graveler.Commit, error) {
+	if err := Validate("repositoryID", repositoryID, ValidateRepositoryID); err != nil {
+		return nil, err
+	}
+	if err := Validate("commitID", commitID, ValidateCommitID); err != nil {
+		return nil, err
+	}
 	return e.store.GetCommit(ctx, repositoryID, commitID)
 }
 
 func (e *EntryCatalog) Dereference(ctx context.Context, repositoryID graveler.RepositoryID, ref graveler.Ref) (graveler.CommitID, error) {
+	if err := Validate("repositoryID", repositoryID, ValidateRepositoryID); err != nil {
+		return "", err
+	}
+	if err := Validate("ref", ref, ValidateRef); err != nil {
+		return "", err
+	}
 	return e.store.Dereference(ctx, repositoryID, ref)
 }
 
 func (e *EntryCatalog) Reset(ctx context.Context, repositoryID graveler.RepositoryID, branchID graveler.BranchID) error {
+	if err := Validate("repositoryID", repositoryID, ValidateRepositoryID); err != nil {
+		return err
+	}
+	if err := Validate("branchID", branchID, ValidateBranchID); err != nil {
+		return err
+	}
 	return e.store.Reset(ctx, repositoryID, branchID)
 }
 
 func (e *EntryCatalog) ResetKey(ctx context.Context, repositoryID graveler.RepositoryID, branchID graveler.BranchID, path Path) error {
+	if err := Validate("repositoryID", repositoryID, ValidateRepositoryID); err != nil {
+		return err
+	}
+	if err := Validate("branchID", branchID, ValidateBranchID); err != nil {
+		return err
+	}
+	if err := Validate("path", path, ValidatePath); err != nil {
+		return err
+	}
+	if err := Validate("path", path.String(), ValidateNonEmptyString); err != nil {
```

Comment:
Why don't you use `ValidatePath` here?

Reply:
If you asked it means I need better naming.
ValidatePath validate that the path is valid - but not required, I will add another function that will call both.